python/ray/data/_internal/datasource/csv_datasource.py

```diff
@@ -37,13 +37,22 @@ def __init__(
         self.parse_options = arrow_csv_args.pop("parse_options", csv.ParseOptions())
         self.arrow_csv_args = arrow_csv_args
 
+        # Initialize projection pushdown attributes
+        self._data_columns = None
+        self._data_columns_rename_map = None
```

Comment:
We should not do that:

 - CSV reader doesn't support projections/predicates, hence
 - We'd not push projections/predicates into it
 - Instead fusion should fuse Read and Project into 1 op (make it exception for fusion and allow to merge Projects into Reads even on RT)